src/bitcoin-cli.cpp

```diff
@@ -255,46 +255,30 @@ class BaseRequestHandler
 /** Process addrinfo requests */
 class AddrinfoRequestHandler : public BaseRequestHandler
 {
-private:
-    int8_t NetworkStringToId(const std::string& str) const
-    {
-        for (size_t i = 0; i < NETWORKS.size(); ++i) {
-            if (str == NETWORKS[i]) return i;
-        }
-        return UNKNOWN_NETWORK;
-    }
-
 public:
     UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
     {
         if (!args.empty()) {
             throw std::runtime_error("-addrinfo takes no arguments");
         }
-        UniValue params{RPCConvertValues("getnodeaddresses", std::vector<std::string>{{"0"}})};
-        return JSONRPCRequestObj("getnodeaddresses", params, 1);
+        return JSONRPCRequestObj("getaddrmaninfo", NullUniValue, 1);
```

Comment:
> how important are IsTerrible addresses, and would it be interesting to have the breakdown with/without that filtering.

`IsTerrible` is used for two things outside of RPCs:
1.) Terrible addresses are not included in answers to GETADDR requests from peers
2.) In case of a conflict (two addrs would go to the same bucket and position in the new tables) the old addr is only overwritten if it's terrible.

Notably, `IsTerrible` isn't used as a filter while making outbound connections, which is the primary function of addrman and the thing users would be most concerned about. That's why I think that the total number of addresses is the better number to report in RPCs.

Reply:
noted. not convinced how desirable the additional code complexity is. would like to hear how other people think about code complexity too before deciding what to do next.